server-phoenix/lib/helios_web/schema/resolvers/event_collection.ex

```diff
@@ -0,0 +1,40 @@
+defmodule HeliosWeb.Schema.Resolvers.EventCollection do
+  @test_query_data %{
+    all: [
+      %{id: 1, source: "github_pull", created_at: ~U[2022-02-16 16:31:17Z]},
+      %{id: 2, source: "github_commit", created_at: ~U[2022-02-16 16:33:17Z]},
+      %{id: 3, source: "slack_message", created_at: ~U[2022-02-16 16:37:17Z]},
+      %{id: 4, source: "github_commit", created_at: ~U[2022-02-17 16:31:17Z]},
+      %{id: 5, source: "github_commit", created_at: ~U[2022-02-17 16:33:17Z]},
+      %{id: 6, source: "slack_message", created_at: ~U[2022-02-17 16:35:17Z]}
+    ],
+    count: %{github_pull: 1, github_commit: 3, slack_message: 2}
+  }
+
+  defp test_query(args) do
+    new_data =
+      case args do
+        %{type: type} ->
+          @test_query_data.all
+          |> Enum.filter(fn event -> event.source == type end)
+
+        %{created_after: created_after} ->
+          @test_query_data.all
+          |> Enum.filter(fn event -> DateTime.compare(event.created_at, created_after) == :gt end)
+
+        %{created_after: created_after, type: type} ->
+          @test_query_data.all
+          |> Enum.filter(fn event -> DateTime.compare(event.created_at, created_after) == :gt end)
+          |> Enum.filter(fn event -> event.source == type end)
+
+        _ ->
+          @test_query_data.all
```

Comment:
I suspect there is a more idiomatic way to write this. Maybe using `if/2` in the pipeline or using function argument matching/extract filter functions. Asking #elixir will probably provide some good options.

Reply:
This is what it looks like with argument pattern matching. I'll see what #elixir also thinks about it. 
![image](https://user-images.githubusercontent.com/67285754/156195275-41111a3a-4438-4e8c-9a6a-d95a377b12d4.png)